imu.c

```diff
@@ -0,0 +1,534 @@
+         * Note2 : A higher number of averaged samples will result in a lower noise level of the signal, but
+         * this has an adverse effect on the power consumed.
+         */
+        config[ACCEL].cfg.acc.bwp = BMI2_ACC_NORMAL_AVG4;
+
+        /* Enable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         * For more info refer datasheet.
+         */
+        config[ACCEL].cfg.acc.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* The user can change the following configuration parameters according to their requirement. */
+        /* Set Output Data Rate */
+        config[GYRO].cfg.gyr.odr = BMI2_GYR_ODR_200HZ;
+
+        /* Gyroscope Angular Rate Measurement Range.By default the range is 2000dps. */
+        config[GYRO].cfg.gyr.range = BMI2_GYR_RANGE_2000;
+
+        /* Gyroscope bandwidth parameters. By default the gyro bandwidth is in normal mode. */
+        config[GYRO].cfg.gyr.bwp = BMI2_GYR_NORMAL_MODE;
+
+        /* Enable/Disable the noise performance mode for precision yaw rate sensing
+         * There are two modes
+         *  0 -> Ultra low power mode(Default)
+         *  1 -> High performance mode
+         */
+        config[GYRO].cfg.gyr.noise_perf = BMI2_POWER_OPT_MODE;
+
+        /* Enable/Disable the filter performance mode where averaging of samples
+         * will be done based on above set bandwidth and ODR.
+         * There are two modes
+         *  0 -> Ultra low power mode
+         *  1 -> High performance mode(Default)
+         */
+        config[GYRO].cfg.gyr.filter_perf = BMI2_PERF_OPT_MODE;
+
+        /* Set the accel and gyro configurations. */
+        rslt = bmi2_set_sensor_config(config, 2, bmi);
+        bmi2_print_error_code(rslt);
+    }
+
+    return rslt;
+}
+
+/*!
+ * @brief This function converts lsb to meter per second squared for 16 bit accelerometer at
+ * range 2G, 4G, 8G or 16G.
+ */
+static float lsb_to_mps2(int16_t val, float g_range, uint8_t bit_width)
+{
+    double power = 2;
+
+    float half_scale = (float)((pow((double)power, (double)bit_width) / 2.0f));
+
+    return (GRAVITY_EARTH * val * g_range) / half_scale;
+}
+
+/*!
+ * @brief This function converts lsb to degree per second for 16 bit gyro at
+ * range 125, 250, 500, 1000 or 2000dps.
+ */
+static float lsb_to_dps(int16_t val, float dps, uint8_t bit_width)
+{
+    double power = 2;
+
+    float half_scale = (float)((pow((double)power, (double)bit_width) / 2.0f));
+
+    return (dps / (half_scale)) * (val);
+}
+
+/*!
+ *  @brief Prints the execution status of the APIs.
+ */
+void bmi2_print_error_code(int8_t rslt)
+{
+    switch (rslt)
+    {
+        case BMI2_OK:
+
+            /* Do nothing */
+            break;
+
+        case BMI2_W_FIFO_EMPTY:
+            NRF_LOG_INFO("Warning [%d] : FIFO empty\r\n", rslt);
```

Comment:
can we replace these all with NRF_LOG_RAW_INFO("[imu] error_code:%d - log message here")